src/CMakeLists.txt

```diff
@@ -49,4 +49,6 @@ endif()
 # Install executable
 # =============================================================================
 install(TARGETS nmodl DESTINATION ${NMODL_INSTALL_DIR_SUFFIX}bin)
-install(FILES nmodl.hpp DESTINATION ${NMODL_INSTALL_DIR_SUFFIX}include)
+if(NOT NMODL_BUILD_WHEEL)
+  install(FILES nmodl.hpp DESTINATION ${NMODL_INSTALL_DIR_SUFFIX}include)
```

Comment:
Why are we only installing `nmodl.hpp` when we're not building a wheel?

Reply:
What's the usecase for distributing them in the wheel (genuinely asking)?

Reply:
I wish `h5py` would distribute the headers for the HDF5 library they install; if they would, I could build python packages that link against the version of HDF5 they use; because they don't I would need to figure out which the correct version is, guess how to build it; and hope I end up with compatible headers. Then I could try to find their `libhdf5.so` and build a package that links to it.

Since HDF5 is a library with a lot of global state, if two applications don't use the same library they can't talk to each other, e.g. pass the integer identifier of a dataset from one library to the next.

For NMODL, I'd just like less special cases. If you don't want headers in the wheel you can strip them at a later stage. Much like the manifest dictates what goes into the source distribution.

Reply:
I mean, the user still then needs to get the paths right, as the wheel is self-contained and doesn't put anything in `/usr/[local]/include`, `pip install` basically just extracts the files from the wheel into `site-packages`.

> If you don't want headers in the wheel you can strip them at a later stage.

Not very easily (i.e. I would need to remove stuff from the already-built wheel) as `scikit-build-core` doesn't have an option to change the `install_manifest.txt`, or exclude any files from it (also the reason why I use the same flag to disable `fmtlib` from installing itself).

All of the above being said, I can leave this header in as its default install location is fine (unlike `fmtlib`).

Reply:
That's unfortunate, maybe it's the lesser evil. I was hoping we could completely eliminate the need for a special wheel mode.

Reply:
After merging #1193, I think we no longer need special cases for what's installed when. The tree is now:
```
build-debug/install
├── bin
│   ├── lexer
│   │   ├── c_lexer
│   │   ├── nmodl_lexer
│   │   └── units_lexer
│   ├── nmodl
│   ├── parser
│   │   ├── c_parser
│   │   ├── nmodl_parser
│   │   └── units_parser
│   └── visitor
│       └── nmodl_visitor
├── include
│   ├── crout
│   │   └── crout.hpp
│   ├── Eigen
...
│   ├── newton
│   │   └── newton.hpp
│   └── nmodl.hpp
├── lib
│   └── nmodl
│       ├── ast.py
│       ├── dsl.py
│       ├── ext
│       │   ├── example
│       │   │   ├── exp2syn.mod
│       │   │   ├── expsyn.mod
│       │   │   ├── hh.mod
│       │   │   └── passive.mod
│       │   └── viz
│       │       ├── css
│       │       │   └── tree.css
│       │       ├── index.html
│       │       └── js
│       │           ├── d3.min.js
│       │           └── tree.js
│       ├── __init__.py
│       ├── _nmodl.cpython-311-x86_64-linux-gnu.so
│       ├── ode.py
│       ├── symtab.py
│       └── visitor.py
└── share
    └── nmodl
        └── nrnunits.lib
```

Reply:
Do we need these two parts in the wheel? They get copied in `nmodl/.data/include` if I don't disable it with the `NMODL_BUILD_WHEEL` flag:

https://github.com/BlueBrain/nmodl/blob/85a4906f3458f29843ba57381a946aa4fccefdb1/src/solver/CMakeLists.txt#L12
https://github.com/BlueBrain/nmodl/blob/85a4906f3458f29843ba57381a946aa4fccefdb1/src/solver/CMakeLists.txt#L18

Reply:
With the right MOD file I can generate code that looks like this:
```
#include <coreneuron/utils/randoms/nrnran123.h>
#include <newton/newton.hpp>


namespace coreneuron {
```
The file `<newton/newton.hpp> is supposed to be provided by NMODL which I think is why you'll find it in the tree. It in turn uses Eigen:
```
#include <crout/crout.hpp>

#include <Eigen/Dense>
#include <Eigen/LU>
```

Therefore when `nrnivmodl` does (the equivalent of)
```
g++ -c kinetic01.cpp -o kinetic01.o
```
it needs to know where to find `<newton/newton.hpp>` and Eigen. The way this is solved is by installing the required files. Now `nrnivmodl` can do:
```
g++ -I ${NMODL_INSTALL_DIR}/include -c kinetic01.cpp -o kinetic01.o
```

Since NMODL generates the calls to the functions inside of `newton.hpp` it seems to convenient to include `newton.hpp` in NMODL rather than ship it with NEURON. This way changing how we generate code that needs to solve a non-linear system is contained inside NMODL. (This isn't true if one uses `scopmath/newton_thread.hpp`; the common case.)

Reply:
When checking the output of `nrnivmodl` we find:
```
/usr/bin/c++ -g  -O0 -fstack-protector   -fopenmp-simd -std=c++17 -DUSE_PYTHON -DNRN_ENABLE_THREADS -DMPI_NO_CPPBIND=1 -DOMPI_SKIP_MPICXX=1 -DMPICH_SKIP_MPICXX=1 -DHAVE_CONFIG_H  -I".." -I.   -I/home/lucg/git/bbp/nrn/build-nmodl/install/include  -I/usr/include -fPIC -c "/home/lucg/git/bbp/nmodl_benchmark/kinetic/x86_64/kinetic01.cpp" -o kinetic01.o
```
Meaning it doesn't in the slightest care where NMODL installs the sources, it'll always use the headers (and version of Eigen) that was installed with NEURON.